lib/vector-common/src/finalizer.rs

```diff
@@ -42,9 +42,22 @@ where
     S: FuturesSet<FinalizerFuture<T>> + Default + Send + Unpin + 'static,
 {
     /// Produce a finalizer set along with the output stream of
-    /// received acknowledged batch identifiers.
+    /// received acknowledged batch identifiers. The output stream will end immediately when
+    /// a shutdown signal is received, or when the sender is dropped
```

Comment:
Is this right about the sender? This implies different behavior than without shutdown below, where the output stream ends after the sender is dropped _and_ the stream is drained.